File: 90-subsets-ii/subsets-ii.java
class Solution {
    void check(int i,List<Integer> sa,List<List<Integer>> ans,int[] nums,int n){
        //List<Integer> a=new ArrayList<>(sa);
        ans.add(new ArrayList<>(sa));
        for(int j=i;j<n;j++){
            if(j==i || nums[j]!=nums[j-1]){
                sa.add(nums[j]);
                //ans.add(sa);
                check(j+1,sa,ans,nums,n);
                sa.remove(sa.size()-1);
            }
        }
    }
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>> ans=new ArrayList<>();
        List<Integer> sa=new ArrayList<>();
        int n=nums.length;
        // ans.add(sa);
        check(0,sa,ans,nums,n);
        return ans;
    }
}

File: 90-subsets-ii/subsets-ii.py
class Solution(object):
    def subsetsWithDup(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """

        def find(i, ans, subans):

            if i == len(nums):
                b = [k for k in subans]
                ans.append(b)
                return
            
            for j in range(i, len(nums)):
                if i == j or nums[j] != nums[j - 1]:
                    subans.append(nums[j])
                    find(j + 1, ans, subans)
                    subans.pop(-1)
        
            find(len(nums), ans, subans)



        nums.sort()
        ans = []
        find(0, ans, [])
        return ans